src/main/java/net/openhft/chronicle/threads/Threads.java

```diff
@@ -201,12 +201,17 @@ static void forEachThread(ExecutorService service, Consumer<Thread> consumer) {
 
             List<Object> objects = listTL.get();
             objects.clear();
-            if (mainLock != null) mainLock.lock();
```

Comment:
why change?

Reply:
SQ could not determine that the lock was always released so I tried to make it more obvious.